forge/db/controllers/AccessToken.js

```diff
@@ -174,15 +174,19 @@ module.exports = {
      * @param {Object} app - The app object
      * @param {string} name - A name for this token
      * @param {Object|number} team - The team object or id
-     * @param {Object|string} project - The project object or id
+     * @param {'application'|'instance'} [autoAssignType] - The type of auto assign (set to `null` or `undefined` to remove auto assign)
+     * @param {Object|string} [autoAssignId] - The auto assign id (set to `null` or `undefined` to remove auto assign)
      * @param {Date|'never'} [expiresAt] - The expiry date. If `undefined`, the token never expires
      */
-    createTokenForTeamDeviceProvisioning: async function (app, name, team, project, expiresAt) {
+    createTokenForTeamDeviceProvisioning: async function (app, name, team, autoAssignType, autoAssignId, expiresAt) {
         const generatedToken = generateToken(32, 'ffadp')
         const scope = ['device:provision', `name:${name}`]
-        const projectId = (project && typeof project === 'object') ? project.id : project
+        const projectId = autoAssignType === 'instance' ? (autoAssignId && typeof autoAssignId === 'object') ? autoAssignId.id : autoAssignId : null
+        const applicationId = autoAssignType === 'application' ? (autoAssignId && typeof autoAssignId === 'object') ? autoAssignId.hashid : autoAssignId : null
```

Comment:
It might just be me, but these double shortcut if statements are REALLY hard to read/understand

Reply:
I assume the are evaluated in to out?
e.g.

```
const projectId = autoAssignType === 'instance' ?
       (autoAssignId && typeof autoAssignId === 'object') ? autoAssignId.id : autoAssignId 
: null
```

Reply:
Fair comment. I have altered this to be more verbose and clear in [code readability: remove double ternary logic](https://github.com/FlowFuse/flowfuse/pull/5394/commits/8d87abe8ba5bfdc9da4b31a68deecb497025f471)
[8d87abe](https://github.com/FlowFuse/flowfuse/pull/5394/commits/8d87abe8ba5bfdc9da4b31a68deecb497025f471)